satellite/metainfo/validation.go

```diff
@@ -189,6 +189,10 @@ func (endpoint *Endpoint) validateCommitSegment(ctx context.Context, req *pb.Seg
 			return Error.New("invalid no order limit for piece")
 		}
 
+		if req.Pointer.SegmentSize > endpoint.rsConfig.MaxSegmentSize.Int64() || req.Pointer.SegmentSize < 0 {
+			return Error.New("segment size %v too large, maximum is %v", req.Pointer.SegmentSize, endpoint.rsConfig.MaxSegmentSize)
```

Comment:
in case of negative `req.Pointer.SegmentSize` this message will be a little bit awkward :)